satpy/scene.py

```diff
@@ -44,63 +44,100 @@
+        new_scn.save_datasets()
+
+    """
+
+    def __init__(self, filenames=None, reader=None, filter_parameters=None, reader_kwargs=None,
                  ppp_config_dir=get_environ_config_dir(),
-                 reader=None,
                  base_dir=None,
                  sensor=None,
                  start_time=None,
                  end_time=None,
-                 area=None,
-                 reader_kwargs=None,
-                 **metadata):
-        """The Scene object constructor.
-
-        Note to load data either `filenames`, `reader`, or a 'base_dir' must
-        be specified. If `filenames` is not specified then `reader` must be
-        specified to search the current directory or `base_dir` if specified.
-        If neither `filenames` nor `reader` is specified then `base_dir` will
-        be used to find any files matching the file patterns for any
-        configured readers. Otherwise the Scene is created with no Readers
-        available meaning Datasets must be added manually:
-
-            scn = Scene(sensor='viirs', start_time=start_time)
+                 area=None):
+        """Initialize Scene with Reader and Compositor objects.
+
+        To load data `filenames` and preferably `reader` must be specified. If `filenames` is provided without `reader`
+        then the available readers will be searched for a Reader that can support the provided files. This can take
+        a considerable amount of time so it is recommended that `reader` always be provided. Note without `filenames`
+        the Scene is created with no Readers available requiring Datasets to be added manually::
+
+            scn = Scene()
             scn['my_dataset'] = Dataset(my_data_array, **my_info)
 
         Args:
-            filenames (iterable): A sequence of files that will be used to load
-                                  data from.
-            ppp_config_dir (str): The directory containing the configuration
-                                  files for satpy.
-            reader: The name of the reader to use for loading the data.
-            base_dir (str): The directory to search for files containing the
+            filenames (iterable or dict): A sequence of files that will be used to load data from. A ``dict`` object
+                                          should map reader names to a list of filenames for that reader.
+            reader (str or list): The name of the reader to use for loading the data or a list of names.
+            filter_parameters (dict): Specify loaded file filtering parameters.
+                                      Shortcut for `reader_kwargs['filter_parameters']`.
+            reader_kwargs (dict): Keyword arguments to pass to specific reader instances.
+            ppp_config_dir (str): The directory containing the configuration files for satpy.
+            base_dir (str): (DEPRECATED) The directory to search for files containing the
```

Comment:
Should we have an alternative indicated in the DEPRECATED parentheses (like the others) ?

Reply:
Yes.